pkg/workflow/providers/multicluster/deploy.go

```diff
@@ -165,13 +183,148 @@ func (t *applyTask) dependents() []string {
 	return dependents
 }
 
+func (t *applyTask) inputs() []string {
+	var inputs []string
+	for _, input := range t.component.Inputs {
+		inputs = append(inputs, t.varKey(input.From))
+	}
+	return inputs
+}
+
+func (t *applyTask) outputs() []string {
+	var outputs []string
+	for _, output := range t.component.Outputs {
+		outputs = append(outputs, t.varKey(output.Name))
+	}
+	return outputs
+}
+
+func (t *applyTask) varKey(v string) string {
+	return fmt.Sprintf("%s/%s/%s", t.placement.Cluster, t.placement.Namespace, v)
+}
+
+func (t *applyTask) patchInput(inputs *ioCache, build valueBuilder) error {
```

Comment:
Function implementation here is a bit redundant. Way to optimize:
1. Create a empty cue.Value x
2. x = x.FillValue(cue.ParsePath(""), comp.Properties)
3. for key, val := range inputs { x = x.FillValue(cue.ParsePath(key), val) }
4. x.UnmarshalTo(comp.Properties)
So that we can avoid explicit json parsing and FillValueByScript, reduce the number of errors we need to handle in this function. BTW, this logic could be separated into some util packages.

Reply:
I remove the marshal process on the first step. However, we can not use a random cue.Value to do the patch, because the runtime is not the same with values from input.

Reply:
I think the input is not necessarily a cue.Value?

Reply:
When we get input from workflowContext, it is value.Value.

Reply:
I see. Well, let's leave the issue here. It is a non-blocking issue for this PR. I will go back and handle it in the future.

Reply:
@Somefive Here's some correction: we are not getting input from workflowContext. We will re-apply healthy components and dynamically compute the output. Using same value.Value's runtime is critical here.